### PR TITLE
Improve footways visibility

### PR DESCRIPTION
**Pull request by pastk**
Needed-for: #2123

Darker farmlands to make footways and roads more visible:
footway

Darker parkings:
parking

Similar used colors comparison (parkings are on left-right edges):
footway

Made steps appear at z14 same as footways, otherwise footways miss connections at z14-15. Made steps rendering a tad thinner also.
steps

### CODE REVIEW

**Comment by pastk** [comment]
Maybe both footways and steps should be moved from z14 to z15, because sometime they connect service roads / driveways which are visible from z15... And footways are more usual in urban areas overloaded with detail anyway.

**Comment by biodranik** [comment]
У пешеходных троп и дорог (белых) точно нет никакой прозрачности?

**Comment by pastk** [comment]
> Maybe both footways and steps should be moved from z14 to z15, because sometime they connect service roads / driveways which are visible from z15... And footways are more usual in urban areas overloaded with detail anyway.

demote-footways

**Comment by biodranik** [comment]
No, zoom levels should not be changed for footpaths. They are very helpful for navigation in the wild.

The problem is in cities because of the density, and it should be solved in a different way.

Also, it's good to keep colors different: many users know that white usually means a good quality footpath, while brown means a potentially dirty track.

**Comment by pastk** [comment]
> No, zoom levels should not be changed for footpaths. They are very helpful for navigation in the wild.
> 

I suggested it exactly because `footway`s are very seldom used for countryside / wild areas, mappers usually use them for urban ones (and its vice versa for `path`s and `track`s which I didn't touch).

If we won't move footways from z14 to z15 then I suggest promoting service roads from z15 to z14. Because at the moment its completely illogical - more prominent ways are not rendered while smaller ones are displayed.

**Comment by pastk** [comment]
And for the rare occasion of an unpaved footway or the one w/ `sac_scale=*` we should consider converting it into the `path` I think: https://github.com/organicmaps/organicmaps/issues/1796#issuecomment-1007758806

**Comment by biodranik** [comment]
Where to test footways: 51.694377, -1.169257

**Comment by pastk** [comment]
> > No, zoom levels should not be changed for footpaths. They are very helpful for navigation in the wild.
> 
> I suggested it exactly because `footway`s are very seldom used for countryside / wild areas, mappers usually use them for urban ones (and its vice versa for `path`s and `track`s which I didn't touch).
> 
> If we won't move footways from z14 to z15 then I suggest promoting service roads from z15 to z14. Because at the moment its completely illogical - more prominent ways are not rendered while smaller ones are displayed.

I have removed the "move footways from z14 to z15" commit from this PR. Its better done separately I think.

File: data/styles/clear/include/Roads.mapcss [hunk 1]
```diff
@@ -804,13 +804,13 @@ line|z13-[highway=proposed]
...
 line|z13-[highway=footway][bridge?]::bridgewhite,
-line|z16-[highway=steps][bridge?]::bridgewhite,
+line|z14-[highway=steps][bridge?]::bridgewhite,
 line|z16-[highway=road][bridge?]::bridgewhite,
 line|z16-[highway=service][bridge?]::bridgewhite
 {casing-linecap: butt;casing-color:@bridge_background;}
 line|z13-[highway=pedestrian][bridge?]::bridgeblack,
 line|z13-[highway=footway][bridge?]::bridgeblack,
-line|z16-[highway=steps][bridge?]::bridgeblack,
+line|z14-[highway=steps][bridge?]::bridgeblack,
```
Comment (biodranik) [inline data/styles/clear/include/Roads.mapcss:813]: Зачем промоутить?

File: data/styles/clear/include/Roads.mapcss [hunk 2]
```diff
@@ -824,7 +824,7 @@ line|z14-[route=hiking],
 line|z14-[highway=bridleway]
 {color: @bridleway;opacity: 1;}
 line|z14-[highway=footway],
-line|z16-[highway=steps]
+line|z14-[highway=steps]
```
Comment (biodranik) [inline data/styles/clear/include/Roads.mapcss:827]: Зачем ступеньки протаскивать? Они же супер мелкие, в городах особенно.

**Comment by biodranik** [comment]
@pastk обновишь, пожалуйста?

File: data/styles/clear/include/Roads.mapcss (hunk 2 above)
Comment (pastk) [reply data/styles/clear/include/Roads.mapcss:827]: зачастую короткие и тогда z16- нормально, но если длинные, то возникает разрыв в дорожках, см. последнюю партию скриншотов в первом посте
(может это не лучший пример, т.к. на сером фоне их плохо видно - могу получше пример привести)

Comment (pastk) [reply data/styles/clear/include/Roads.mapcss:827]: https://www.openstreetmap.org/way/374402168
в скриншотах вот это место, если в аппе позумить, то хорошо видно, как тропа разрывается

Comment (pastk) [reply data/styles/clear/include/Roads.mapcss:827]: вот ещё примеры длинных лестниц
https://www.openstreetmap.org/way/454828316
https://www.openstreetmap.org/way/377222081
тут куча их в округе https://www.openstreetmap.org/way/219812662
когда они пропадают, а дорожки вокруг остаются, то картина пешей проходимости сильно искажённая получается

Comment (biodranik) [reply data/styles/clear/include/Roads.mapcss:827]: Но включение на z14 ради немногочисленных длинных ступенек сильно "замусорит" карту и возможно даже замедлит отрисовку. Предлагаю лучше сделать отдельный issue для этого. По-хорошему, надо указывать "рекомендованный" для отрисовки зум, но грамотно фильтровать и не рисовать то, что будет слишком мелко. Т.е. по сути динамические правила. @vng или мы уже в коде фильтруем короткие линейные объекты?

**Pull request merged**